src/Features/Core/Portable/Completion/Providers/AbstractTypeImportCompletionProvider.cs

```diff
@@ -223,14 +232,32 @@ internal override async Task<CompletionChange> GetChangeAsync(Document document,
 
                 return CompletionChange.Create(Utilities.Collapse(newText, builder.ToImmutableAndFree()));
             }
-            else
+
+            static bool ShouldCompleteWithFullyQualifyTypeName(Document document)
             {
-                // For workspace that doesn't support document change, e.g. DebuggerIntellisense
-                // we complete the type name in its fully qualified form instead.
-                var fullyQualifiedName = $"{containingNamespace}.{completionItem.DisplayText}";
-                var change = new TextChange(completionListSpan, fullyQualifiedName);
+                var workspace = document.Project.Solution.Workspace;
 
-                return CompletionChange.Create(change);
+                // Certain types of workspace don't support document change, e.g. DebuggerIntellisense
+                if (!workspace.CanApplyChange(ApplyChangesKind.ChangeDocument))
+                {
+                    return true;
+                }
+
+                // During an EnC session, adding import is not supported.
+                var encService = workspace.Services.GetService<IDebuggingWorkspaceService>()?.EditAndContinueServiceOpt;
```

Comment:
I think this should move to FeatureService ?

Reply:
@tmat What do you think?

Reply:
What should?

Reply:
Should we return "CodeFix and Refactorings not available" in the ITextBufferSupportsFeatureService during EnC?